usr/src/cmd/mdb/common/mdb/mdb_ctf.c

```diff
@@ -1468,29 +1468,52 @@ vread_helper(mdb_ctf_id_t modid, char *modbuf,
 		return (mdb_ctf_member_iter(modid, member_cb, &mbr));
 
 	case CTF_K_UNION:
+		mbr.m_modbuf = modbuf;
+		mbr.m_tgtbuf = tgtbuf;
+		mbr.m_tgtid = tgtid;
+		mbr.m_flags = flags;
+		mbr.m_tgtname = typename;
 
 		/*
-		 * Unions are a little tricky. The only time it's truly
-		 * safe to read in a union is if no part of the union or
-		 * any of its component types have changed.  The correct
-		 * use of this feature is to read the containing structure,
-		 * figure out which component of the union is valid, compute
-		 * the location of that in the target and then read in
-		 * that part of the structure.
+		 * Not all target union members need to be present in the
+		 * mdb type. If there is only a single union member in the
+		 * mdb type, its actual type does not need to match with
```

Comment:
OK, I think I see what you're saying. Basically if the target member is there then the encapsulating union and its sizing doesn't matter, as we're just reading it into the target type, that right?